packages/chat-headless-react/tests/headlessProvider.test.tsx

```diff
@@ -0,0 +1,63 @@
+import { render } from "@testing-library/react";
+import {
+  ChatHeadlessProvider,
+  ConversationState,
+  HeadlessConfig,
+} from "../src";
+import { renderToString } from "react-dom/server";
+import { useChatState } from "../src/useChatState";
+
+it("only fetches session storage on client-side render", async () => {
+  const win = window;
+  Object.defineProperty(win, "sessionStorage", {
+    value: {
+      ...win.sessionStorage,
+      getItem: (_: string): string => {
+        return JSON.stringify({
+          messages: [{ text: "foobar", source: "BOT" }],
+          isLoading: false,
+          canSendMessage: false,
+        } satisfies ConversationState);
+      },
+    },
+  });
+  const windowSpy = jest
+    .spyOn(window, "window", "get")
+    .mockImplementation(() => win);
+  const config: HeadlessConfig = {
+    botId: "123",
+    apiKey: "1234",
+    saveToSessionStorage: true,
+  };
+  const str = () =>
+    renderToString(
+      <ChatHeadlessProvider config={config}>
+        <TestComponent />
+      </ChatHeadlessProvider>
+    );
+  expect(windowSpy).not.toHaveBeenCalled();
```

Comment:
since we haven't call renderToString before this, this expect wouldn't be asserting anything. We should move this expect to AFTER line 40 `container.innerHTML = str();`

Reply:
whoops, good catch